lib/web_ui/test/engine/view_embedder/embedding_strategy/custom_element_embedding_strategy_test.dart

```diff
@@ -121,4 +121,33 @@ void doTests() {
           reason: 'Should be injected `nextTo` the passed element.');
     });
   });
+
+  group('context menu', () {
+    setUp(() {
+      target = createDomElement('this-is-the-target');
+      domDocument.body!.append(target);
+      strategy = CustomElementEmbeddingStrategy(target);
+      strategy.initialize();
+    });
+
+    tearDown(() {
+      target.remove();
+    });
+
+    test('disableContextMenu and enableContextMenu can toggle the context menu', () {
+      expect(strategy.contextMenuEnabled, isTrue);
+
+      strategy.disableContextMenu();
+      expect(strategy.contextMenuEnabled, isFalse);
```

Comment:
Is there anything else I could expect here, like could I somehow verify that the event listener was added to the dom element?

Reply:
You could add your own `contextmenu` Event handler to the root DOM element of the strategy in the test, and check  if [`isDefaultPrevented`](https://developer.mozilla.org/en-US/docs/Web/API/Event/defaultPrevented) has been set to `true`. (IIRC, once you prevent the default in a handler, the event will be modified to set `defaultPrevented` to true.)

In order to programmatically trigger the event, you should be able to [`dispatchEvent`](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/dispatchEvent) a [`contextmenu` event](https://developer.mozilla.org/en-US/docs/Web/API/Element/contextmenu_event) using the root DOM element of the strategy.

Reply:
There are some examples in our code base. E.g. https://github.com/flutter/engine/blob/7bff5dd115d794200266c523f332c408c1da25c8/lib/web_ui/test/raw_keyboard_test.dart#L57

You don't have to create an event listener on the root element. Just check `defaultPrevented` on the event you dispatched.

Reply:
That worked, thank you all for the guidance here!  I think the tests are much more robust this way, and I don't have to use `@visibleForTest`.

Reply:
> Just check defaultPrevented on the event you dispatched.

Damn this approach is good! Thanks for the tests @justinmc!